### PR TITLE
Bazel 1.0 support

### LINKED ISSUE #22: Bazel 1.0.0 support
Build currently breaks on recently released Bazel 1.0.0, see https://buildkite.com/bazel/rules-proto-grpc-rules-proto-grpc/builds/91. Pending issues:

- [x] Update gRPC to fix proto provider issue (not determined if this is fixed upstream yet)
- [x] Update rules_scala to fix create_provider issue. Resolved upstream in https://github.com/bazelbuild/rules_scala/issues/513
- [x] Update all dependencies
- [x] Rename rules_proto_grpc_dependencies to rules_proto_grpc_repos
- [ ] Write release notes
- [x] ~~Add default WORKSPACE rules to each language docs~~
- [ ] Try updating C# packages, otherwise bump to 1.x.0
- [x] Fix grpc-gateway
- [x] Fix rust rules
- [x] Drop gRPC.js

Other linked issues are on the [1.0.0 milestone](https://github.com/rules-proto-grpc/rules_proto_grpc/milestone/2).

Changes are being made on the [bazel-1.0-support branch](https://github.com/rules-proto-grpc/rules_proto_grpc/tree/bazel-1.0-support).

### PR DESCRIPTION
**Pull request by aaliddell**
TODO:

- [x] Update all dependencies
- [x] Write release notes
- [x] Replace rules_pip with rules_python
- [x] ~~Fix scala~~
- [x] Fix rust

Closes #22, #14, #8, #10, #4, #23, #30

### CODE REVIEW

File: repositories.bzl [hunk 1]
```diff
@@ -35,29 +38,51 @@ VERSIONS = {
...
+
+    # Misc
     "com_github_bazelbuild_buildtools": {
         "type": "github",
         "org": "bazelbuild",
         "repo": "buildtools",
-        "ref": "6415663945d3248207da955aafa1fa2af1a0f2ed",
-        "sha256": "d1e28237d1f4c2255c504246b4f3fd36f74d590f2974491b4399a84c58b495ed",
+        "ref": "0.28.0",
+        "sha256": "d1e28237d1a4c2255c504246b4f3fd36f74d590f2974491b4399a84c58b495ed",
```
Comment (plule-ansys) [inline repositories.bzl]: bazel reports `5ec71602e9b458b01717fab1d37492154c1c12ea83f881c745dbd88e9b2098d8` for this one

**Comment by plule-ansys** [comment]
For info, to get the C# generated code to be compatible with the nuget packages, I had to upgrade the grpc and protobuf packages to their latest versions:
google.protobuf: 3.10.1
grpc, grpc.core, grpc.core.api: 2.25.0

File: repositories.bzl (hunk 1 above)
Comment (aaliddell) [reply repositories.bzl]: Good spot, fixed, thanks

**Comment by aaliddell** [comment]
All languages except Scala are now working and at this point I'm considering bumping Scala support off the 1.0.0 target and disabling its tests in CI, as it's taking a massively disproportionate amount of my time to troubleshoot and subsequently delaying this merge. Scala support can be fixed in 1.1.0 if that's the decision made.

> For info, to get the C# generated code to be compatible with the nuget packages, I had to upgrade the grpc and protobuf packages to their latest versions:
google.protobuf: 3.10.1
grpc, grpc.core, grpc.core.api: 2.25.0

I can't ever get the C# rules to work, let alone update the packages. Could you provide the changes as a PR based on this branch, so I can see if any tests work? Or see if https://github.com/rules-proto-grpc/rules_proto_grpc/pull/34 is the same?

**Comment by plule-ansys** [comment]
> I can't ever get the C# rules to work, let alone update the packages. Could you provide the changes as a PR based on this branch, so I can see if any tests work? Or see if #34 is the same?

Unfortunately I can’t manage to build any of the examples here, and I’m pretty much lost in the whole .net frameworks, rules and nuget bazel files generations, so I’m not yet able to provide any proper PR. It’s the same kind of changes as #34. I’m trying to better understand the C# rules and nuget2bazel stuff, hopefully I’ll be able to contribute here.

**Pull request merged**